### PR TITLE
OAuth2ResourceServerSpec should allow its ServerAuthenticationConverter to be configurable

### PR DESCRIPTION
**Pull request by edeandrea**
Fixes gh-6186

### CODE REVIEW

File: config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java [hunk 1]
```diff
@@ -429,6 +447,38 @@ ReactiveAuthenticationManager authenticationManager() {
...
 
+	@EnableWebFlux
+	@EnableWebFluxSecurity
+	static class CustomBearerTokenServerAuthenticationConverter {
+		@Bean
+		SecurityWebFilterChain springSecurity(ServerHttpSecurity http) throws Exception {
+			// @formatter:off
+			http
+				.authorizeExchange()
+					.anyExchange().access(authorizationManager())
```
Comment (jzheaux) [inline config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java]: Is there a reason you need this extra instrumentation, as opposed to something like 

```suggestion
					.anyExchange().access("SCOPE_message:read")
```

If we can, it's nice to be able to reduce the number of moving parts in a test.

Comment (edeandrea) [reply config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java]: You are correct I can switch this test. I will issue a new commit that changes this.

**Comment by edeandrea** [comment]
@jzheaux I pushed a new commit and rebased so the commit message is better. Sorry about that!

**Comment by edeandrea** [comment]
Also - any chance of getting this in pre 5.2?

**Comment by jgrandja** [comment]
@edeandrea This can only get into 5.2 since [bearerTokenConverter(...)](https://github.com/spring-projects/spring-security/pull/6190/files#diff-f6e82f2a5486ce7fe3f6438343fc05f0R927) is new

**Comment by edeandrea** [comment]
@jgrandja [bearerTokenConverter(...)](
https://github.com/spring-projects/spring-security/pull/6190/files#diff-f6e82f2a5486ce7fe3f6438343fc05f0R927) is what I added in the PR. There aren't any new classes or anything added in this PR - just a change to the configuration DSL.

**Comment by jzheaux** [comment]
Thanks for the PR, @edeandrea! This is now merged into `master`.

**Pull request merged**